components/brave_shields/browser/ad_block_service.h

```diff
@@ -110,7 +110,8 @@ class AdBlockService {
                           bool* did_match_rule,
                           bool* did_match_exception,
                           bool* did_match_important,
-                          std::string* mock_data_url);
+                          std::string* mock_data_url,
+                          std::string* rewritten_url);
```

Comment:
i think we should stop adding new params to this method and introduce a structure instead. Otherwise, it goes out of control - there is no single line of comment describing what is for what, and every new change touches a dozen of irrelevant files

Reply:
cc @bridiver

Reply:
Ideally we just need https://github.com/brave/brave-browser/issues/5461, which I'm hoping to get to soon:tm:

or definitely at least before adding any further arguments to these methods